cli/src/semdep/parse_lockfile.py

```diff
@@ -491,6 +571,33 @@ def parse_dep(s: str) -> FoundDependency:
     yield from (parse_dep(dep) for dep in deps)
 
 
+def parse_requirements(
+    lockfile_text: str, manifest_text: Optional[str]
+) -> Iterator[FoundDependency]:
+    deps = [
+        (i, l.split("==")) for i, l in enumerate(lockfile_text.split("\n")) if "==" in l
```

Comment:
I added a function to remove comments in case people decide to put them on the same line as a package. None of the automated tools do this. Otherwise yes, any line that does not contain an `==` is either a comment, a newline, or was a non-pinned package, which is now filtered at the top.